hi-ml-cpath/src/health_cpath/utils/deepmil_utils.py

```diff
@@ -38,6 +38,26 @@ def set_module_gradients_enabled(model: nn.Module, tuning_flag: bool) -> None:
         params.requires_grad = tuning_flag
 
 
+class Resnet18(ImageNetEncoder):
+    def __init__(self, tile_size: int, n_channels: int = 3) -> None:
+        super().__init__(resnet18, tile_size, n_channels, apply_imagenet_preprocessing=True)
+
+
+class Resnet18_NoPreproc(ImageNetEncoder):
+    def __init__(self, tile_size: int, n_channels: int = 3) -> None:
+        super().__init__(resnet18, tile_size, n_channels, apply_imagenet_preprocessing=False)
+
+
+class Resnet50(ImageNetEncoder):
+    def __init__(self, tile_size: int, n_channels: int = 3) -> None:
+        super().__init__(resnet50, tile_size, n_channels, apply_imagenet_preprocessing=True)
+
+
+class Resnet50_NoPreproc(ImageNetEncoder):
+    def __init__(self, tile_size: int, n_channels: int = 3) -> None:
+        super().__init__(resnet50, tile_size, n_channels, apply_imagenet_preprocessing=False)
+
```

Comment:
Are these classes used in any config test? if we want tests to reflect the latest status of the codebase they should

Reply:
No, I didn't want to make a lot of changes (I will need to update the tests and quite a lot of config files).
I just added these new classes so that we can use them in cpath. I know that it's not ideal cause it introduces some code duplication, but we can maybe address that in a separate PR with a cleaner refactoring. I will add a ticket to the Board